src/fgimic/integral.f90

```diff
@@ -103,17 +103,19 @@ subroutine integrate_current(this, mol, xdens, spinn)
         xsum3=0.d0
         psum3=0.d0
         nsum3=0.d0
-!$OMP PARALLEL PRIVATE(i,j,k,r,rr,xsum,psum,nsum) &
+
+!$OMP PARALLEL DEFAULT(NONE) &
+!$OMP PRIVATE(i,j,k,r,rr,xsum,psum,nsum) &
 !$OMP PRIVATE(jt,w,jp,tt,jvec) &
-!$OMP SHARED(xsum2,psum2,nsum2) &
-!$OMP SHARED(p1,p2,p3,this,center,spin,bb,normal,mol,xdens,lo,hi) &
-!$OMP REDUCTION(+:xsum3,psum3,nsum3)
+!$OMP SHARED(p1,p2,p3,this,center,spin,bb,normal,mol,xdens,lo,hi,bound) &
```

Comment:
here mol, and xdens are defined but they are not used in the loop - why do you have to do this?

Reply:
Much better - thank you ...

Reply:
All variables inside the parallel region need to be defined.

Reply:
I know this has also been done before but I was wondering why.

Reply:
Threads spawn in the parallel region but the actual concurrent work is done inside the DO-loop.

Reply:
I thought everything in the "parallel region" is inside the loop. This is apparently wrong. How do you decide on what needs to be in the parallel region?

Reply:
The parallel region could be just the loop. But creating and closing parallel regions costs time so you don't want to create them too often either. As small as possible, as large as necessary.